etcd_test.go

```diff
@@ -0,0 +1,30 @@
+// +build integration
+
+package etcd
+
+import (
+	"testing"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/giantswarm/microstorage/storagetest"
+)
+
+func Test_Storage(t *testing.T) {
+	etcdConfig := clientv3.Config{
+		Endpoints: []string{"http://127.0.0.1:2379"},
+	}
+	etcdClient, err := clientv3.New(etcdConfig)
+	if err != nil {
+		t.Fatal("expected", nil, "got", err)
+	}
+
+	config := DefaultConfig()
+	config.EtcdClient = etcdClient
+
+	storage, err := New(config)
+	if err != nil {
+		t.Fatal("expected", nil, "got", err)
+	}
+
+	storagetest.Test(t, storage)
+}
```

Comment:
I had issues with key sanitisation and returning internal errors, but nothing major.